skill-tree /src/main/java/com/RDS/skilltree/Skill/SkillsController.java

```diff
@@ -0,0 +1,69 @@
+package com.RDS.skilltree.Skill;
+
+import com.RDS.skilltree.Exceptions.NoEntityException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.coyote.Response;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.ObjectUtils;
+import org.springframework.web.bind.annotation.*;
+
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+
+@RestController
+@Slf4j
+@RequestMapping("/v1/skills")
+public class SkillsController {
+    private final SkillsService skillsService;
+    public SkillsController(SkillsService skillsService){
+        this.skillsService = skillsService;
+    }
+
+    @PostMapping("/")
+    public ResponseEntity<?> createSkill(@RequestBody(required = true) SkillDRO skillDRO){
+        if (skillDRO.getCreatedBy() == null || skillDRO.getType() == null || skillDRO.getName() == null){
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body("CreatedBy, Type and Name are mandatory values, anyone cannot be null");
+        }
+        try {
+            return ResponseEntity.status(HttpStatus.CREATED).body(skillsService.createSkill(skillDRO));
+        } catch(DataIntegrityViolationException ex){
+            return ResponseEntity.status(HttpStatus.CONFLICT)
+                    .body("Cannot create entry for Skill as Skill name is duplicate");
+        }
+    }
+
+    @GetMapping("/")
+    public Page<SkillDTO> getAllSkills(
+            @RequestParam(value = "page", defaultValue = "0") int page,
+            @RequestParam(value = "size", defaultValue = "100") int size) {
+        Pageable pageable = PageRequest.of(page, size);
+        return skillsService.getAllSkills(pageable);
```

Comment:
404 case missing if no skills found

Reply:
@heyrandhir IMO this response is having much context about the number of skills returned.
```
{
    "content": [],
    "pageable": {
        "sort": {
            "empty": true,
            "unsorted": true,
            "sorted": false
        },
        "offset": 0,
        "pageNumber": 0,
        "pageSize": 100,
        "paged": true,
        "unpaged": false
    },
    "last": true,
    "totalPages": 0,
    "totalElements": 0,
    "first": true,
    "size": 100,
    "number": 0,
    "sort": {
        "empty": true,
        "unsorted": true,
        "sorted": false
    },
    "numberOfElements": 0,
    "empty": true
}
```